### DISCUSSION TITLE
My Chatbot is remembering old files, how can I delete the memory?

### DISCUSSION THREAD

**Discussion by MrMrch** [FlowiseAI/Flowise#979] [Q&A] [answered]
I need to reset the memory of the chatbot, it keeps using an older version of the file which was a test, and now I can't make it stop. 
Im using the QA Conversational Retrieval as per this tutorial https://www.youtube.com/watch?v=riXpu1tHzl0&ab_channel=DaveEbbelaar

**Comment by trendion** [comment]
same problem here

**Comment by HenryHengZJ** [comment]
Have you specified filter or namespace or combination of both for the Pinecone node?

**Reply by MrMrch** [reply]
Mmh I think only the namespace, but I'm not at my pc right now

**Reply by MrMrch** [reply]
@HenryHengZJ  sorry so I added a new file instead of the previous one, but still not seeing any change. The new informations are not received or used when asking the questions

**Comment by LeandroTecware** [comment]
I have the same problem. As a first question, from a new chat it is possible to notice that the chat has accumulated content or returns an error message:

**This model's maximum context length is 4097 tokens. However, you requested 4140 tokens (3140 in the messages, 1000 in the completion). Please reduce the length of the messages or completion.**

**Comment by rvy474** [comment, marked as answer]
Upserting new data does not mean the old data is deleted. You will first need to delete the old data. Below is some python code that I ran on Google Collab.

#install pinecone client
!pip install pinecone-client

#get the project connected
import pinecone
pinecone.init(api_key="_your key_", environment="_your environment_")
index = pinecone.Index("_your index_")

#delete the index
index.delete(delete_all=True, namespace='')

If you have not added a namespace then use '' else use the actual name of the namespace.

Deleting the old data, saving your chat flow and then upserting the new data should solve the problem

**Comment by HenryHengZJ** [comment]
you can either delete the old data, or store the new data to a new namespace/collection, basically concept of another folder. That way you can prevent the mix of knowledge file/data

**Comment by mwmdev** [comment]
What you need is a [Record Manager](https://docs.flowiseai.com/integrations/langchain/record-managers) node in your flow, in particular setting the « Cleanup » option in additional parameters.
